pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -3,12 +3,27 @@
 import signal
 from subprocess import call
 
+from pytorch_lightning.plugins.environments import SLURMEnvironment
+from pytorch_lightning.utilities.imports import _fault_tolerant_training
+
 log = logging.getLogger(__name__)
 
 
-class SLURMConnector:
+class SignalConnector:
     def __init__(self, trainer):
         self.trainer = trainer
+        self.trainer._should_gracefully_terminate = False
+
+    def register_signal_handlers(self):
+        cluster_env = getattr(self.trainer.training_type_plugin, "cluster_environment", None)
```

Comment:
noob q: is this not defined for all TTPs? why?

Reply:
It is not decided for SingleDevice ahah